GTest not compiling

Hello,
today i wanted to use gtest with my eclipse / mingw makefile project.
On my machine(Win7) is a working mingw toolchain installed.
I followed the instructions on 

> http://www.codeproject.com/Articles/811934/Cplusplus-unit-test-start-guide-how-to-set-up-Goog

to set up gtest. But when i try to run the tests i get the following errors:
```
Info: Internal Builder is used for build
g++ "-IC:\\Users\\XXX\\eclipse_playground_workspace\\unitttest\\include" "-IC:\\Users\\XXX\\eclipse_playground_workspace\\unitttest\\gtest_src" -O0 -g3 -Wall -c -fmessage-length=0 -o "test\\gtest_main.o" "..\\test\\gtest_main.cpp" 
g++ "-IC:\\Users\\XXX\\eclipse_playground_workspace\\unitttest\\include" "-IC:\\Users\\XXX\\eclipse_playground_workspace\\unitttest\\gtest_src" -O0 -g3 -Wall -c -fmessage-length=0 -o "test\\test_factorial.o" "..\\test\\test_factorial.cpp" 
g++ -O0 -g3 -Wall -c -fmessage-length=0 -o "src\\factorial.o" "..\\src\\factorial.cpp" 
g++ "-IC:\\Users\\XXX\\eclipse_playground_workspace\\unitttest\\gtest_src" -O0 -g3 -Wall -c -fmessage-length=0 -o "gtest_src\\gtest\\gtest-all.o" "..\\gtest_src\\gtest\\gtest-all.cc" 
..\gtest_src\gtest\gtest-all.cc: In constructor 'testing::internal::Mutex::Mutex()':
..\gtest_src\gtest\gtest-all.cc:8791:45: error: cannot convert 'CRITICAL_SECTION* {aka _CRITICAL_SECTION*}' to '_RTL_CRITICAL_SECTION*' in initialization
       critical_section_(new CRITICAL_SECTION) {
                                             ^
..\gtest_src\gtest\gtest-all.cc:8792:48: error: cannot convert '_RTL_CRITICAL_SECTION*' to 'LPCRITICAL_SECTION {aka _CRITICAL_SECTION*}' for argument '1' to 'void InitializeCriticalSection(LPCRITICAL_SECTION)'
   ::InitializeCriticalSection(critical_section_);
                                                ^
..\gtest_src\gtest\gtest-all.cc: In destructor 'testing::internal::Mutex::~Mutex()':
..\gtest_src\gtest\gtest-all.cc:8802:46: error: cannot convert '_RTL_CRITICAL_SECTION*' to 'PCRITICAL_SECTION {aka _CRITICAL_SECTION*}' for argument '1' to 'void DeleteCriticalSection(PCRITICAL_SECTION)'
     ::DeleteCriticalSection(critical_section_);
                                              ^
..\gtest_src\gtest\gtest-all.cc: In member function 'void testing::internal::Mutex::Lock()':
..\gtest_src\gtest\gtest-all.cc:8810:43: error: cannot convert '_RTL_CRITICAL_SECTION*' to 'LPCRITICAL_SECTION {aka _CRITICAL_SECTION*}' for argument '1' to 'void EnterCriticalSection(LPCRITICAL_SECTION)'
   ::EnterCriticalSection(critical_section_);
                                           ^
..\gtest_src\gtest\gtest-all.cc: In member function 'void testing::internal::Mutex::Unlock()':
..\gtest_src\gtest\gtest-all.cc:8820:43: error: cannot convert '_RTL_CRITICAL_SECTION*' to 'LPCRITICAL_SECTION {aka _CRITICAL_SECTION*}' for argument '1' to 'void LeaveCriticalSection(LPCRITICAL_SECTION)'
   ::LeaveCriticalSection(critical_section_);
                                           ^
..\gtest_src\gtest\gtest-all.cc: In member function 'void testing::internal::Mutex::ThreadSafeLazyInit()':
..\gtest_src\gtest\gtest-all.cc:8841:27: error: cannot convert 'CRITICAL_SECTION* {aka _CRITICAL_SECTION*}' to '_RTL_CRITICAL_SECTION*' in assignment
         critical_section_ = new CRITICAL_SECTION;
                           ^
..\gtest_src\gtest\gtest-all.cc:8842:54: error: cannot convert '_RTL_CRITICAL_SECTION*' to 'LPCRITICAL_SECTION {aka _CRITICAL_SECTION*}' for argument '1' to 'void InitializeCriticalSection(LPCRITICAL_SECTION)'
         ::InitializeCriticalSection(critical_section_);
                                                      ^
..\gtest_src\gtest\gtest-all.cc: In static member function 'static void testing::internal::ThreadLocalRegistryImpl::StartWatcherThreadFor(DWORD)':
..\gtest_src\gtest\gtest-all.cc:9044:21: error: '::OpenThread' has not been declared
     HANDLE thread = ::OpenThread(SYNCHRONIZE | THREAD_QUERY_INFORMATION,
                     ^

13:31:56 Build Finished (took 3s.12ms)`
```

I tried to build it directly without eclipse. So i used cmake with mingw makefile configuration. But results stay the same when executing mingw32-make
Am i missing something?
Do i need special packages from the mingw installation?

Thanks in advance for your help.

--- a/googletest/cmake/internal_utils.cmake
+++ b/googletest/cmake/internal_utils.cmake
@@ -46,7 +46,9 @@ endmacro()
 # Google Mock.  You can tweak these definitions to suit your need.  A
 # variable's value is empty before it's explicitly assigned to.
 macro(config_compiler_and_linker)
-  if (NOT gtest_disable_pthreads)
+  # Note: pthreads on MinGW is not supported, even if available
+  # instead, we use windows threading primitives
+  if (NOT gtest_disable_pthreads AND NOT MINGW)
     # Defines CMAKE_USE_PTHREADS_INIT and CMAKE_THREAD_LIBS_INIT.
     find_package(Threads)
   endif()

--- a/googletest/include/gtest/internal/gtest-port.h
+++ b/googletest/include/gtest/internal/gtest-port.h
@@ -396,10 +396,16 @@
 #  include <io.h>
 # endif
 // In order to avoid having to include <windows.h>, use forward declaration
-// assuming CRITICAL_SECTION is a typedef of _RTL_CRITICAL_SECTION.
+#if GTEST_OS_WINDOWS_MINGW
+// MinGW defined _CRITICAL_SECTION and _RTL_CRITICAL_SECTION as two
+// separate (equivalent) structs, instead of using typedef
+typedef struct _CRITICAL_SECTION GTEST_CRITICAL_SECTION;
+#else
+// Assume CRITICAL_SECTION is a typedef of _RTL_CRITICAL_SECTION.
 // This assumption is verified by
 // WindowsTypesTest.CRITICAL_SECTIONIs_RTL_CRITICAL_SECTION.
-struct _RTL_CRITICAL_SECTION;
+typedef struct _RTL_CRITICAL_SECTION GTEST_CRITICAL_SECTION;
+#endif
 #else
 // This assumes that non-Windows OSes provide unistd.h. For OSes where this
 // is not the case, we need to include headers that provide the functions
@@ -1693,7 +1699,7 @@ class GTEST_API_ Mutex {
   // by the linker.
   MutexType type_;
   long critical_section_init_phase_;  // NOLINT
-  _RTL_CRITICAL_SECTION* critical_section_;
+  GTEST_CRITICAL_SECTION* critical_section_;
 
   GTEST_DISALLOW_COPY_AND_ASSIGN_(Mutex);
 };

--- a/googletest/test/gtest-port_test.cc
+++ b/googletest/test/gtest-port_test.cc
@@ -1295,9 +1295,16 @@ TEST(WindowsTypesTest, HANDLEIsVoidStar) {
   StaticAssertTypeEq<HANDLE, void*>();
 }
 
+#if GTEST_OS_WINDOWS_MINGW
+TEST(WindowsTypesTest, _CRITICAL_SECTIONIs_CRITICAL_SECTION) {
+  StaticAssertTypeEq<CRITICAL_SECTION, _CRITICAL_SECTION>();
+}
+#else
 TEST(WindowsTypesTest, CRITICAL_SECTIONIs_RTL_CRITICAL_SECTION) {
   StaticAssertTypeEq<CRITICAL_SECTION, _RTL_CRITICAL_SECTION>();
 }
+#endif
+
 #endif  // GTEST_OS_WINDOWS
 
 }  // namespace internal

--- a/googletest/test/gtest_unittest.cc
+++ b/googletest/test/gtest_unittest.cc
@@ -442,7 +442,7 @@ class FormatEpochTimeInMillisAsIso8601Test : public Test {
     // tzset() distinguishes between the TZ variable being present and empty
     // and not being present, so we have to consider the case of time_zone
     // being NULL.
-#if _MSC_VER
+#if _MSC_VER || GTEST_OS_WINDOWS_MINGW
     // ...Unless it's MSVC, whose standard library's _putenv doesn't
     // distinguish between an empty and a missing variable.
     const std::string env_var =

